pkg/detectors/buildkite/v1/buildkite.go

```diff
@@ -79,3 +71,29 @@ func (s Scanner) Type() detectorspb.DetectorType {
 func (s Scanner) Description() string {
 	return "Buildkite is a platform for running fast, secure, and scalable continuous integration pipelines. Buildkite API tokens can be used to access and modify pipeline data and configurations."
 }
+
+func VerifyBuildKite(ctx context.Context, client *http.Client, secret string) (bool, error) {
+	// create a request
+	req, err := http.NewRequestWithContext(ctx, "GET", "https://api.buildkite.com/v2/access-token", nil)
+	if err != nil {
+		return false, err
+	}
+
+	// add authorization header
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", secret))
+
+	res, err := client.Do(req)
+	if err != nil {
+		return false, err
+	}
+	defer func() {
+		_, _ = io.Copy(io.Discard, res.Body)
+		_ = res.Body.Close()
+	}()
+
+	if res.StatusCode == http.StatusOK {
+		return true, nil
```

Comment:
I think it's worth including the `scopes` in extraData.

https://buildkite.com/docs/apis/rest-api/access-token#get-the-current-token